scipy/_lib/_numpy_compat.py

```diff
@@ -45,3 +49,236 @@ def _assert_warns(warning_class, func, *args, **kw):
                 raise AssertionError("First warning for %s is not a "
                         "%s( is %s)" % (func.__name__, warning_class, l[0]))
         return result
+
+
+if NumpyVersion(np.__version__) >= '1.12.0.dev':
```

Comment:
This will break for 1.12 dev versions without ``suppress_warnings``, so better make this a feature check (just try import it from ``numpy.testing``)